### PR TITLE
Add types

### PR DESCRIPTION
**Pull request by tylerodonnell**
This PR adds a `index.d.ts` file to be used for typescript.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      exp: number,
+      iat: number,
+      iss: string
+      jti?: number,
+      scp?: Array<String>
+      sub: string
+      uid?: number,
+
+      // Custom JWT claims
+      aid?: string,
```
Comment (tylerodonnell) [inline index.d.ts]: @vscerchia Now that I've moved this into `@articulate/authentic`, having our custom JWT claims here seems wrong. Should we extend the JWT interface in our code and add our custom claims there?

Also, what are your thoughts on adding `[key: string]: any` to the JWT interface? I think we should so anyone using this package can use their custom claims.

Comment (vscerchia) [reply index.d.ts]: Hey @tylerodonnell, I agree we should remove the custom claims and add `[key: string]: any` here.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      // Standard JWT claims
+      aud: string,
+      cid?: number,
+      exp: number,
+      iat: number,
+      iss: string
+      jti?: number,
+      scp?: Array<String>
+      sub: string
+      uid?: number,
```
Comment (vscerchia) [inline index.d.ts]: TIL both commas and semicolons are valid delimiters in interface declarations

Comment (tylerodonnell) [reply index.d.ts]: I actually prefer the no comma look. Thanks for pointing out the inconsistency :)

File: index.d.ts (hunk 1 above)
Comment (tylerodonnell) [reply index.d.ts]: 👍 Done in a163b241799f8e1de5625e7f63c910a1af1f7e4d

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    verify?: VerifyOpts
+  }
+
+  interface Validator {
+    (token: string): Promise<Authentic.JWT>
+  }
+
+  function Authentic(opts: AuthenticOpts): Validator
+
+  namespace Authentic {
```
Comment (vscerchia) [inline index.d.ts:32]: Is there any reason for the namespace here?

**Review by vscerchia — APPROVED** [review]
Left one question, but I think this looks good otherwise.

Not sure if you saw this, but I found this template on the typescript site: https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-function-d-ts.html which I think would be want we want here except for the fact that we're exporting the function using `module.exports` instead of `export = `. This seems like a good first pass, though, since I don't think we're interested in changing the actual code if we can avoid it right now.

File: index.d.ts (hunk 3 above)
Comment (tylerodonnell) [reply index.d.ts:32]: I had to use a namespace so we're able to use a namespace import

```js
import * as Authentic from "@articulate/authentic";
```

This allows users to still be able to do

```js
import * as Authentic from "@articulate/authentic"
Authentic() // function
Authentic.JWT // interface
```
```js
import authentic from "@articulate/authentic"
authentic() // function
```
```js
import { JWT } from "@articulate/authentic"
JWT // interface
```

Comment (tylerodonnell) [reply index.d.ts:32]: Without the namespace, you'll get an error in typscript: `Cannot invoke an expression whose type lacks a call signature.`

**Comment by tylerodonnell** [comment]
@vscerchia Yeah I'm trying to leave the code alone with this PR. I'd love to make that suggestion in another version 👍

**Review by vscerchia — APPROVED** [review]

**Pull request merged**